roborazzi-idea-plugin/src/main/kotlin/com/github/takahirom/roborazzi/idea/preview/PreviewViewModel.kt

```diff
@@ -0,0 +1,275 @@
+package com.github.takahirom.roborazzi.idea.preview
+
+import com.github.takahirom.roborazzi.idea.settings.AppSettingsState
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.impl.source.tree.CompositeElement
+import com.intellij.psi.util.PsiTreeUtil
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import kotlinx.coroutines.yield
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtFunction
+import java.io.File
+
+class PreviewViewModel {
+
+  var coroutineScope = MainScope()
+  val imagesStateFlow = MutableStateFlow<List<Pair<String, Long>>>(listOf())
+  private val lastEditingFileName = MutableStateFlow<String?>(null)
+  val statusText = MutableStateFlow("No images found")
+  private val _dropDownUiState = MutableStateFlow(ActionToolbarUiState())
+  val dropDownUiState = _dropDownUiState.asStateFlow()
+  val shouldSeeIndex = MutableStateFlow(-1)
+  private var updateListJob: Job? = null
+  private val gradleTask = RoborazziGradleTask()
+
+  fun onRefreshClicked(project: Project) {
+    roborazziLog("onRefreshClicked")
+    refreshList(project)
+  }
+
+  fun onInit(project: Project) {
+    roborazziLog("onInit")
+    refreshList(project)
+  }
+
+  fun onSelectedFileChanged(project: Project) {
+    roborazziLog("onSelectedFileChanged")
+    coroutineScope.launch {
+      updateListJob?.cancel()
+      refreshListProcess(project)
+      selectListIndexByCaret(project)
+      fetchTasks(project)
+    }
+  }
+
+  fun onCaretPositionChanged(project: Project) {
+    roborazziLog("onCaretPositionChanged")
+    coroutineScope.launch {
+      updateListJob?.cancel()
+      refreshListProcess(project)
+      selectListIndexByCaret(project)
+      fetchTasks(project)
+    }
+  }
+
+  fun executeTaskByName(project: Project, taskName: String) {
+    roborazziLog("Executing task '$taskName'...")
+    _dropDownUiState.update { currentUiState ->
+      currentUiState.copy(flag = ActionToolbarUiState.Flag.LOADING)
+    }
+    gradleTask.executeTaskByName(project, taskName) {
+      _dropDownUiState.update { currentUiState ->
+        currentUiState.copy(flag = ActionToolbarUiState.Flag.IDLE)
```

Comment:
Could we add `refreshList(project)` here to update images after the Gradle task?